src/syscheckd/syscheck_audit.c

```diff
@@ -1451,65 +1446,31 @@ int audit_health_check(int audit_socket) {
         w_cond_wait(&audit_hc_started, &audit_hc_mutex);
     w_mutex_unlock(&audit_hc_mutex);
 
-    // Create a file
-    fp = fopen(AUDIT_HEALTHCHECK_FILE, "w");
-
-    if(!fp) {
-        mdebug1(FIM_AUDIT_HEALTHCHECK_FILE);
-        goto exit_err;
-    }
-    fclose(fp);
-
-    /*
-     * This is a workaround to fix the whodata mode init when is restarted unexpectedly
-     * *********************************************************************************
-     */
-
-    sleep(1);
-
-    fp = fopen(AUDIT_HEALTHCHECK_FILE, "w");
-
-    if(!fp) {
-        mdebug1(FIM_AUDIT_HEALTHCHECK_FILE);
-        goto exit_err;
-    }
-    fclose(fp);
-
-    mdebug2(FIM_HEALTHCHECK_WAIT_CREATE);
-    sleep(1);
+    // Generate open events until they get picked up
```

Comment:
I'd like to add some changes from here:
- Remove the goto exit_err
- Delete file FIM_HEALTHCHECK_CHECK_RULE in case of error
- In case `fopen()` fails, we go out with an error. We have to retry the operation in that case.
- Delete the message FIM_HEALTHCHECK_CREATE_RECEIVE and print only FIM_HEALTHCHECK_SUCCESS in case of success.
- Add a message in case of error to replace "error: audit_health_check_creation"

Something like that:
```c
    // Generate open events until they get picked up
    do {
        fp = fopen(AUDIT_HEALTHCHECK_FILE, "w");

        if(!fp) {
            mdebug1(FIM_AUDIT_HEALTHCHECK_FILE);
            retval = -1;
        } else {
            fclose(fp);
        }

        sleep(1);
    } while (!audit_health_check_creation && --timer > 0);

    if (!audit_health_check_creation) {
        mdebug1(FIM_HEALTHCHECK_CREATE_FAILED);
        retval = -1;
    } else {
        mdebug1(FIM_HEALTHCHECK_SUCCESS);
        retval = 0;
    }

    // Delete healthcheck file
    unlink(AUDIT_HEALTHCHECK_FILE);

    if(audit_delete_rule(AUDIT_HEALTHCHECK_DIR, AUDIT_HEALTHCHECK_KEY) <= 0){
        mdebug1(FIM_HEALTHCHECK_CHECK_RULE);    // LCOV_EXCL_LINE
    }
    hc_thread_active = 0;

    return retval;
}
```
Remember to modify the unit tests as necessary.
Thanks!